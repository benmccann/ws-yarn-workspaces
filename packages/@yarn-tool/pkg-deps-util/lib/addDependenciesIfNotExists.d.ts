import { IPackageJson } from '@ts-type/package-dts';
import { IOptionsInstallDepsFromWorkspaces } from './types';
import { IPackageJsonDependenciesField } from '@ts-type/package-dts/lib/package-json/types';
export declare function addDependencies(pkg: IPackageJson, name: string, semver: string, options?: IOptionsInstallDepsFromWorkspaces, override?: boolean): {
    pkg: IPackageJson<any>;
    bool: boolean;
};
export declare function _add_to_deps_field(pkg: IPackageJson, field: IPackageJsonDependenciesField, name: string, semver: string, override: boolean, bool: boolean): boolean;
export declare function addDependenciesIfNotExists(pkg: IPackageJson, name: string, semver: string, options?: IOptionsInstallDepsFromWorkspaces): {
    pkg: IPackageJson<any>;
    bool: boolean;
};
