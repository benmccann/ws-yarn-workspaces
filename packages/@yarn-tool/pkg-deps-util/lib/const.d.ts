export declare const enum EnumInstallTypesErrorCode {
    SUCCESS = 0,
    NOT_EXISTS = 1,
    DEPRECATED = 2,
    SKIP = 3
}
