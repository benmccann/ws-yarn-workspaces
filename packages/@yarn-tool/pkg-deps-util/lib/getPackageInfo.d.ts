import { IParsePackageName } from '@yarn-tool/npm-package-arg-util/lib/types';
import { AbbreviatedVersion } from 'package-json';
export declare function getPackageInfo(packageName: string | IParsePackageName, excludeVersion?: boolean): import("bluebird")<AbbreviatedVersion>;
