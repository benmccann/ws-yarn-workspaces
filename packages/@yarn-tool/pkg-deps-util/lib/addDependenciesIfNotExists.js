"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.addDependenciesIfNotExists = exports.addDependencies = exports._add_to_deps_field = void 0;
var pkg_deps_add_1 = require("@yarn-tool/pkg-deps-add");
Object.defineProperty(exports, "_add_to_deps_field", { enumerable: true, get: function () { return pkg_deps_add_1._add_to_deps_field; } });
Object.defineProperty(exports, "addDependencies", { enumerable: true, get: function () { return pkg_deps_add_1.addDependencies; } });
Object.defineProperty(exports, "addDependenciesIfNotExists", { enumerable: true, get: function () { return pkg_deps_add_1.addDependenciesIfNotExists; } });
//# sourceMappingURL=addDependenciesIfNotExists.js.map