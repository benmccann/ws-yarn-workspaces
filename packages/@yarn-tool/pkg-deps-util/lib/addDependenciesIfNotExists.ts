import { IPackageJson } from '@ts-type/package-dts';
import { IOptionsInstallDepsFromWorkspaces } from './types';
import { IPackageJsonDependenciesField } from '@ts-type/package-dts/lib/package-json/types';

export function addDependencies(pkg: IPackageJson,
	name: string,
	semver: string,
	options: IOptionsInstallDepsFromWorkspaces = {},
	override?: boolean,
)
{
	let bool: boolean = null;

	if (options.dev)
	{
		bool = _add_to_deps_field(pkg,
			'devDependencies',
			name,
			semver,
			override,
			bool,
		);
	}

	if (options.peer)
	{
		bool = _add_to_deps_field(pkg,
			'peerDependencies',
			name,
			semver,
			override,
			bool,
		);
	}

	if (options.optional)
	{
		bool = _add_to_deps_field(pkg,
			'optionalDependencies',
			name,
			semver,
			override,
			bool,
		);
	}

	if (bool === null)
	{
		bool = _add_to_deps_field(pkg,
			'dependencies',
			name,
			semver,
			override,
			bool,
		);
	}

	return {
		pkg,
		bool,
	}
}

export function _add_to_deps_field(pkg: IPackageJson,
	field: IPackageJsonDependenciesField,
	name: string,
	semver: string,
	override: boolean,
	bool: boolean,
)
{
	const record = pkg[field] ?? {};

	if (record[name] !== semver)
	{
		if (!record[name]?.length || override === true)
		{
			pkg[field] ??= {};
			pkg[field][name] = semver;

			bool = false;
		}
		else
		{
			bool ??= true;
		}
	}

	return bool
}

export function addDependenciesIfNotExists(pkg: IPackageJson,
	name: string,
	semver: string,
	options: IOptionsInstallDepsFromWorkspaces = {},
)
{
	return addDependencies(pkg, name, semver, options)
}
