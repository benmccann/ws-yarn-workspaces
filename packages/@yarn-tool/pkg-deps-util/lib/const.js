"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnumInstallTypesErrorCode = void 0;
var EnumInstallTypesErrorCode;
(function (EnumInstallTypesErrorCode) {
    EnumInstallTypesErrorCode[EnumInstallTypesErrorCode["SUCCESS"] = 0] = "SUCCESS";
    EnumInstallTypesErrorCode[EnumInstallTypesErrorCode["NOT_EXISTS"] = 1] = "NOT_EXISTS";
    EnumInstallTypesErrorCode[EnumInstallTypesErrorCode["DEPRECATED"] = 2] = "DEPRECATED";
    EnumInstallTypesErrorCode[EnumInstallTypesErrorCode["SKIP"] = 3] = "SKIP";
})(EnumInstallTypesErrorCode || (exports.EnumInstallTypesErrorCode = EnumInstallTypesErrorCode = {}));
//# sourceMappingURL=const.js.map