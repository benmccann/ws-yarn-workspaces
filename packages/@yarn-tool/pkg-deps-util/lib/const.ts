export const enum EnumInstallTypesErrorCode
{
	NOT_EXISTS = 1,
	DEPRECATED = 2,

	SKIP = 3,
}
