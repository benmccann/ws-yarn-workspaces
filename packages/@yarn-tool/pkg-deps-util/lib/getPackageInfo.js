"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getPackageInfo = getPackageInfo;
const packageNameToTypes_1 = require("@yarn-tool/npm-package-arg-util/lib/packageNameToTypes");
const core_1 = require("@yarn-tool/pkg-version-query/lib/core");
const package_json_1 = require("package-json");
function getPackageInfo(packageName, excludeVersion) {
    var _a, _b;
    const data = typeof packageName === 'string' ? (0, packageNameToTypes_1.packageNameToTypes)(packageName) : packageName;
    let version = 'latest';
    if (!excludeVersion && ((_a = data.semver) === null || _a === void 0 ? void 0 : _a.length) > 0) {
        version = (_b = data.semver) !== null && _b !== void 0 ? _b : 'latest';
    }
    return (0, core_1._queryVersion)(data.name, {
        version,
        //fullMetadata: true,
    })
        .catch(package_json_1.VersionNotFoundError, () => {
        return (0, core_1._queryVersion)(data.name, {
            version: 'latest',
            //fullMetadata: true,
        });
    })
        .catch(package_json_1.PackageNotFoundError, err => void 0);
}
//# sourceMappingURL=getPackageInfo.js.map