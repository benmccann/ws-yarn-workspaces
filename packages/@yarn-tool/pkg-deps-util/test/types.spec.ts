import { packageNameToTypes } from '@yarn-tool/npm-package-arg-util/lib/packageNameToTypes';

describe(`packageNameToTypes`, () =>
{

	test(`@next/typescript@^1.3.1`, () =>
	{

		let actual = packageNameToTypes(`@next/typescript@^1.3.1`);

		expect(actual).toHaveProperty('name', '@types/next__typescript');

		expect(actual).toMatchSnapshot();

	});

})
