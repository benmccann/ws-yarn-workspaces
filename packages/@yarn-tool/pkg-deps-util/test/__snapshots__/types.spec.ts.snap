// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`packageNameToTypes @next/typescript@^1.3.1 1`] = `
Object {
  "name": "@types/next__typescript",
  "result": Object {
    "escapedName": "@next%2ftypescript",
    "fetchSpec": "^1.3.1",
    "gitCommittish": undefined,
    "gitRange": undefined,
    "name": "@next/typescript",
    "raw": "@next/typescript@^1.3.1",
    "rawSpec": "^1.3.1",
    "registry": true,
    "saveSpec": null,
    "scope": "@next",
    "type": "range",
    "where": undefined,
  },
  "scope": "@types",
  "semver": "^1.3.1",
  "subname": "next__typescript",
}
`;
